### DISCUSSION TITLE
Exit when last tab closed?

### DISCUSSION THREAD

**Discussion by Kelstertx** [sumatrapdfreader/sumatrapdf#2533] [General]
I searched and didn't find anything on this topic....

I can't believe there isn't a setting to allow Sumatra to exit the program when the last tab is closed.  I have been waiting years for that feature... Am I missing it somehow?  Like maybe in the Advanced settings file?

**Comment by kjk** [comment]
`q` does exact that: closes a tab and exits program if that's a last tab

**Comment by GitHubRulesOK** [comment]
In untabbed mode all windows are singletons so closing last window closes program Q is the keyboard equivalent

In Tabbed mode as mentioned you can use q with the single tab, It is ever so simple to close ALL tabs since windows apps should close using F4 there is also CTRL+Q as an alternative.

Closing last tab in tabs mode is designed to simply go back to either the start tab or frequently read without asking are you wanting to close application

**Comment by thomas-mc-work** [comment]
Wouldn't it be possible to let <kbd>ctrl</kbd> + <kbd>w</kbd> do the same thing when tabs are disabled? The purpose is that I don't want to have an MDI interface, but only SDI. So if there is no document - then there shall be no application window.

**Reply by doubletrouble023** [reply]
Yes, this!
I did not know about Q closing until finding this thread. I always close with `ctrl `+ `w ` and I find it annoying that I can't close the program itself the same way when theres no more open documents

**Reply by viadee-BiM** [reply]
You can achieve this behaviour by adding the following to the [advanced options file](https://www.sumatrapdfreader.org/docs/Customizing-keyboard-shortcuts):

```
Shortcuts [
    [
        Cmd = CmdCloseCurrentDocument
        Key = Ctrl + w
    ]
]
```

Now Ctrl+W acts the same way as just pressing "q" (cf. https://www.sumatrapdfreader.org/docs/Commands) :).

**Reply by doubletrouble023** [reply]
Whoa! you're a wizard mate!  Thanks, that did the trick.
I love software that allows for customization!

**Reply by viadee-BiM** [reply]
Happy to help, I was struggling with the same thing :)!

All the kudos go out to the SumatraPDF developers and community ❤️!

**Reply by thomas-mc-work** [reply]
Thanks a lot @viadee-BiM!

For other users: Just be sure to replace the existing (probably empty) `Shortcuts` section somewhere in the middle of the config file. In the first try I've just appended it at the end which was then removed on the next application start.
